test/schemas/testpilot.js

```diff
@@ -44,154 +54,51 @@ describe("The 'testpilot' Schema", function() {
-    ping.payload.timestamp = new Date();
-    ping.payload.events[0].timestamp = new Date().toISOString();
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required field 'timestamp' is a Date ISO string", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.timestamp = new Date().toISOString();
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
+  requiredStrings.forEach((field) => {
+    it(`should reject if required field ${field} is missing`, function(done) {
+      const ping = clone(samplePing);
+      delete ping[field];
+      Joi.validate(ping, testPilotSchema, function(err) {
+        expect(err).to.exist;
+        done();
+      });
     });
-  });
-  it("should reject if required field 'version' is missing", function(done) {
-    const ping = clone(samplePing);
-    delete ping.payload.version;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required field 'version' is malformed", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.version = 12345; // version is an AMO version string, not an integer
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required event field 'event' is missing", function(done) {
-    const ping = clone(samplePing);
-    delete ping.payload.events[0].event;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required event field 'event' is malformed", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.events[0].event = 1;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required event field 'object' is missing", function(done) {
-    const ping = clone(samplePing);
-    delete ping.payload.events[0].object;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required event field 'object' is malformed", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.events[0].object = 1;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
+    it(`should reject if required field ${field} is empty`, function(done) {
+      const ping = clone(samplePing);
+      ping[field] = "";
+      Joi.validate(ping, testPilotSchema, function(err) {
+        expect(err).to.exist;
+        done();
+      });
     });
   });
-  it("should reject if required event field 'timestamp' is missing", function(done) {
+  it("should reject if required field client_time is missing", function(done) {
```

Comment:
Wouldn't this test be covered by the above test called `should reject if required field ${field} is missing`